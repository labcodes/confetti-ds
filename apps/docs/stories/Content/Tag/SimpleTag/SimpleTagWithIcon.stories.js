import React from "react";

import { ICON_TYPES, TAG_COLORS } from "../../../../confetti-ds/src/constants";
import { SimpleTag as Component } from "../../../../confetti-ds/src";

export default {
  title: "Content/Tag/Simple Tag",
  component: Component,
  argTypes: {
    color: {
      control: { type: "select", options: TAG_COLORS },
    },
    skin: {
      control: { type: "inline-radio" },
    },
    icon: {
      control: { type: "select", options: ICON_TYPES },
    },
    thumbSrc: {
      control: false,
    },
  },
};

export const SimpleTagWithIcon = (args) => {
  return (
    <React.Fragment>
      <Component {...args} />
    </React.Fragment>
  )
};

SimpleTagWithIcon.args = {
  text: "demo tag",
  icon: "coin",
};
