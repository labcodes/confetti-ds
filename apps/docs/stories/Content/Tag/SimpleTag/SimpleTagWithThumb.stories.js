import React from "react";

import { TAG_COLORS } from "../../../../confetti-ds/src/constants";
import { SimpleTag as Component } from "../../../../confetti-ds/src";

export default {
  title: "Content/Tag/Simple Tag",
  component: Component,
  argTypes: {
    color: {
      control: { type: "select", options: TAG_COLORS },
    },
    skin: {
      control: { type: "inline-radio" },
    },
    icon: {
      control: false, 
    },
    thumbSrc: {
      control: false,
    },
  },
};

export const SimpleTagWithThumb = (args) => {
  return (
    <React.Fragment>
      <Component {...args} />
    </React.Fragment>
  )
};

SimpleTagWithThumb.args = {
  text: "demo tag",
  thumbSrc: "http://avatars3.githubusercontent.com/u/1887591",
};
