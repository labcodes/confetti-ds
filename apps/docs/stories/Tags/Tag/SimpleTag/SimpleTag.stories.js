import React from "react";

import { TAG_COLORS } from "../../../../../../packages/confetti-ds/src/constants";
import { SimpleTag as Component } from "../../../../../../packages/confetti-ds/src";

export default {
  title: "Tags/Tag/Simple Tag",
  component: Component,
  argTypes: {
    color: {
      control: { type: "select", options: TAG_COLORS },
    },
    skin: {
      control: { type: "inline-radio" },
    },
    icon: {
      control: false,
    },
    thumbSrc: {
      control: false,
    },
  },
};

export const SimpleTag = (args) => {
  return (
    <React.Fragment>
      <Component {...args} />
    </React.Fragment>
  );
};

SimpleTag.args = {
  text: "demo tag",
};
