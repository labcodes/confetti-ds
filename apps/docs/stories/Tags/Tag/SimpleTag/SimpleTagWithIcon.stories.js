import React from "react";

import {
  ICON_TYPES,
  TAG_COLORS,
} from "../../../../../../packages/confetti-ds/src/constants";
import { SimpleTag as Component } from "../../../../../../packages/confetti-ds/src";

export default {
  title: "Tags/Tag/Simple Tag",
  component: Component,
  argTypes: {
    color: {
      control: { type: "select", options: TAG_COLORS },
    },
    skin: {
      control: { type: "inline-radio" },
    },
    icon: {
      control: { type: "select", options: ICON_TYPES },
    },
    thumbSrc: {
      control: false,
    },
  },
};

export const SimpleTagWithIcon = (args) => {
  return (
    <React.Fragment>
      <Component {...args} />
    </React.Fragment>
  );
};

SimpleTagWithIcon.args = {
  text: "demo tag",
  icon: "coin",
};
