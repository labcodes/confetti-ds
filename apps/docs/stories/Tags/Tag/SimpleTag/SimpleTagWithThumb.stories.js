import React from "react";

import { TAG_COLORS } from "../../../../../../packages/confetti-ds/src/constants";
import { SimpleTag as Component } from "../../../../../../packages/confetti-ds/src";

export default {
  title: "Tags/Tag/Simple Tag",
  component: Component,
  argTypes: {
    color: {
      control: { type: "select", options: TAG_COLORS },
    },
    skin: {
      control: { type: "inline-radio" },
    },
    icon: {
      control: false,
    },
    thumbSrc: {
      control: false,
    },
  },
};

export const SimpleTagWithThumb = (args) => {
  return (
    <React.Fragment>
      <Component {...args} />
    </React.Fragment>
  );
};

SimpleTagWithThumb.args = {
  text: "demo tag",
  thumbSrc: "http://avatars3.githubusercontent.com/u/1887591",
};
