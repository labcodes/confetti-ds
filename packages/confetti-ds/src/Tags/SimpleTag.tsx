import React, { useMemo } from "react";
import { isEmpty } from "lodash";

import AbstractTag from "./AbstractTag";
import Icon from "../Icon";
import { IconTypes, TagTypes } from "../constants";

type BaseSimpleTagProps = {
  /** This is the Tag's text. */
  text: string;
  /** Type of the icon to be rendered. Won't render an icon if not passed to the component. Can't have both 'icon' and 'thumbSrc' at the same time. */
  icon?: IconTypes;
  /** Sets Tag's color. */
  color?: TagTypes;
  /** Skin of the rendered Tag. */
  skin?: "pale" | "vivid";
  /** Sets an outline style. */
  isOutline?: boolean;
};

export type SimpleTagProps =
  | (BaseSimpleTagProps & {
      /** Source of the thumbnail to be rendered. Won't render a thumbnail if not passed to the component. Can't have both 'icon' and 'thumbSrc' at the same time. */
      thumbSrc?: undefined;
      /** Source of the thumbnail to be rendered. Won't render a thumbnail if not passed to the component. Can't have both 'icon' and 'thumbSrc' at the same time. */
      thumbAlt?: undefined;
    })
  | (BaseSimpleTagProps & {
      /** Source of the thumbnail to be rendered. Won't render a thumbnail if not passed to the component. Can't have both 'icon' and 'thumbSrc' at the same time. */
      thumbSrc: string;
      /** Source of the thumbnail to be rendered. Won't render a thumbnail if not passed to the component. Can't have both 'icon' and 'thumbSrc' at the same time. */
      thumbAlt: string;
    });

export default function SimpleTag({
  text,
  icon,
  color,
  skin = "pale",
  isOutline = false,
  thumbSrc = "",
  thumbAlt = "",
}: SimpleTagProps) {
  const thumb = () =>
    thumbSrc.length ? (
      <img className="lab-tag__thumb" src={thumbSrc} alt={thumbAlt} />
    ) : undefined;

  const handleIcon = () =>
    icon ? <Icon type={icon} color="black-75" className="lab-tag--left-icon" /> : undefined;

  const checkThumbAndIcon = () => {
    const errorMessage =
      "`SimpleTag` can't be initialized with both `thumb` and `icon` props.";
    if (!isEmpty(thumbSrc) && !isEmpty(icon)) {
      throw new Error(errorMessage);
    }
  };

  useMemo(() => {
    checkThumbAndIcon();
  }, [icon, thumbSrc]);

  return (
    <AbstractTag
      text={text}
      className={`${icon ? ` lab-tag--has-left-icon` : ""}${
        thumbSrc ? ` lab-tag--has-thumb` : ""
      }`}
      isOutline={isOutline}
      skin={skin}
      color={color}
      renderPrefix={handleIcon() || thumb()}
    />
  );
}
