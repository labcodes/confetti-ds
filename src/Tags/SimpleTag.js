import React from "react";
import PropTypes from "prop-types";
import { isEmpty } from "lodash";

import AbstractTag from "./AbstractTag";
import Icon from "../Icon";
import { TAG_COLORS } from "../constants";

export default class SimpleTag extends React.Component {
  static propTypes = {
    /** This is the Tag's text. */
    text: PropTypes.string.isRequired,
    /** Source of the thumb to be rendered. Won't render a thumb if not passed to the component. */
    thumbSrc: PropTypes.string,
    /** Type of the icon to be rendered. Won't render an icon if not passed to the component. */
    icon: PropTypes.string,
    /** Sets Tag's color. */
    color: PropTypes.oneOf(TAG_COLORS),
    /** Skin of the the rendered Tag. */
    skin: PropTypes.string,
    /** Sets an outline style. */
    isOutline: PropTypes.bool,
  };

  static defaultProps = {
    thumbSrc: "",
    icon: "",
    isOutline: false,
    skin: "pale",
    color: "",
  };

  constructor(props) {
    super(props);
    this.checkThumbAndIcon();
  }

  componentDidUpdate() {
    this.checkThumbAndIcon();
  }

  thumb = () => {
    const { thumbSrc } = this.props;
    return thumbSrc ? (
      <img className="lab-tag__thumb" src={thumbSrc} alt="" />
    ) : undefined;
  };

  icon = () => {
    const { icon } = this.props;
    return icon ? (
      <Icon
        type={icon}
        color="black-75"
        size="petit"
        className="lab-tag--left-icon"
      />
    ) : undefined;
  };

  checkThumbAndIcon() {
    const errorMessage =
      "`SimpleTag` can't be initialized with both `thumb` and `icon` props.";
    const { thumbSrc, icon } = this.props;
    if (!isEmpty(thumbSrc) && !isEmpty(icon)) {
      throw new Error(errorMessage);
    }
  }

  render() {
    const { text, thumbSrc, icon, color, skin, isOutline } = this.props;
    return (
      <AbstractTag
        text={text}
        className={`${icon ? ` lab-tag--has-left-icon` : ""}${
          thumbSrc ? ` lab-tag--has-thumb` : ""
        }`}
        isOutline={isOutline}
        skin={skin}
        color={color}
        renderPrefix={this.icon() || this.thumb()}
      />
    );
  }
}
