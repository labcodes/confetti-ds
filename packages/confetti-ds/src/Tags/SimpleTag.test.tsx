import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import SimpleTag from "./SimpleTag";

describe("SimpleTag", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<SimpleTag text="Test render SimpleTag" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders as expected with outline as true", async () => {
    const renderedComponent = renderer
      .create(<SimpleTag text="Test render outline SimpleTag" isOutline />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const wrapper = shallow(
      <SimpleTag text="Test outline SimpleTag" isOutline />
    ).html();
    expect(wrapper).toContain("lab-tag--outline");
  });

  it("renders as expected with a vivid skin", async () => {
    const renderedComponent = renderer
      .create(<SimpleTag text="Test render vivid SimpleTag" skin="vivid" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const wrapper = shallow(
      <SimpleTag text="Test vivid SimpleTag" skin="vivid" />
    ).html();
    expect(wrapper).toContain("lab-tag--vivid");
  });

  it("renders as expected with a purple color", async () => {
    const renderedComponent = renderer
      .create(<SimpleTag text="Test render purple SimpleTag" color="purple" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const wrapper = shallow(
      <SimpleTag text="Test vivid SimpleTag" skin="vivid" />
    ).html();
    expect(wrapper).toContain("lab-tag--vivid");
  });

  it("renders as expected with a magnifying-glass icon", async () => {
    const renderedComponent = renderer
      .create(
        <SimpleTag
          text="Test render SimpleTag with magnifying-glass icon"
          icon="magnifying-glass"
        />
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const wrapper = shallow(
      <SimpleTag
        text="Test SimpleTag with magnifying-glass icon"
        icon="magnifying-glass"
      />
    ).html();
    expect(wrapper).toContain("lab-icon--magnifying-glass");
  });

  it("renders as expected with a thumb", async () => {
    const renderedComponent = renderer
      .create(
        <SimpleTag
          text="Test render SimpleTag with thumb"
          thumbSrc="fake-thumb"
          thumbAlt="fake-thumb-alt"
        />
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const wrapper = shallow(
      <SimpleTag
        text="Test SimpleTag with thumb"
        thumbSrc="fake-thumb"
        thumbAlt="fake-thumb-alt"
      />
    ).html();
    expect(wrapper).toContain(`src="fake-thumb"`);
  });

  it("does not render if passing both `thumb` and `icon` props", async () => {
    expect(() => {
      shallow(
        <SimpleTag
          text="Test to not render SimpleTag with with thumb and icon"
          icon="magnifying-glass"
          thumbSrc="fake-thumb"
          thumbAlt="fake-thumb-alt"
        />
      );
    }).toThrow(
      "`SimpleTag` can't be initialized with both `thumb` and `icon` props."
    );
  });
});
