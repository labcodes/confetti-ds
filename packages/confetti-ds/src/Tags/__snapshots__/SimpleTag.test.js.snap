// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleTag renders as expected with a magnifying-glass icon 1`] = `
<span
  className="lab-tag  lab-tag--has-left-icon lab-tag--pale"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <span
    className="lab-icon lab-icon--magnifying-glass lab-icon--black-75 lab-icon--petit lab-tag--left-icon"
  />
  Test render SimpleTag with magnifying-glass icon
</span>
`;

exports[`SimpleTag renders as expected with a purple color 1`] = `
<span
  className="lab-tag  lab-tag--purple-pale"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  Test render purple SimpleTag
</span>
`;

exports[`SimpleTag renders as expected with a thumb 1`] = `
<span
  className="lab-tag  lab-tag--has-thumb lab-tag--pale"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <img
    alt=""
    className="lab-tag__thumb"
    src="fake-thumb"
  />
  Test render SimpleTag with thumb
</span>
`;

exports[`SimpleTag renders as expected with a vivid skin 1`] = `
<span
  className="lab-tag  lab-tag--vivid"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  Test render vivid SimpleTag
</span>
`;

exports[`SimpleTag renders as expected with outline as true 1`] = `
<span
  className="lab-tag  lab-tag--outline lab-tag--pale"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  Test render outline SimpleTag
</span>
`;

exports[`SimpleTag renders with base props 1`] = `
<span
  className="lab-tag  lab-tag--pale"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  Test render SimpleTag
</span>
`;
